shared/multisig.py

```diff
@@ -889,10 +889,94 @@ def update_contents(self):
         MenuItem('Delete', f=ms_wallet_delete, arg=ms),
         MenuItem('Coldcard Export', f=ms_wallet_ckcc_export, arg=ms),
         MenuItem('Electrum Wallet', f=ms_wallet_electrum_export, arg=ms),
+        MenuItem('Address Explorer', f=multisig_address_explorer, arg=ms)
     ]
 
     return rv
 
+def make_redeem(M, keys, path):
+    # Construct a redeem script, and ordered list of pubkeys
+    N = len(keys)
+
+    # derive pubkeys (see BIP 67: <https://github.com/bitcoin/bips/blob/master/bip-0067.mediawiki>)
+    pubkeys = []
+    for xfp, node in keys:
+        depth = node.depth()
+        for p in path[depth:]: # derive entire path (in-place)
+            node.derive(p)
+        pubkeys.append(node.public_key())
+    pubkeys.sort()
+
+    # construct redeem script
+    mm = [80 + M] if M <= 16 else [1, M]
+    nn = [80 + N] if N <= 16 else [1, N]
+    rv = bytes(mm)
+    for pk in pubkeys:
+        rv += bytes([len(pk)]) + pk
+    rv += bytes(nn + [0xAE])
+
+    return rv, pubkeys
+
+def ms_path_mapper(idx, is_change, coin, addr_fmt):
+    def HARD(n=0):
+        return 0x80000000 | n
+    # currently supported address format derivations
+    d = {
+        AF_P2SH: ([HARD(45), is_change, idx], "m/45'/{change}/{idx}".format(change=is_change, idx=idx)),
+        AF_P2WSH_P2SH: (
+            [HARD(48), HARD(coin), HARD(0), HARD(1), is_change, idx],
+            "m/48'/{coin}'/0'/1'/{change}/{idx}".format(coin=coin, change=is_change, idx=idx)
+        ),
+        AF_P2WSH: (
+            [HARD(48), HARD(coin), HARD(0), HARD(2), is_change, idx],
+            "m/48'/{coin}'/0'/2'/{change}/{idx}".format(coin=coin, change=is_change, idx=idx)
+        )
+    }
```

Comment:
Two thirds of this work is thrown out after a few more lines. Little tiny micro had to calculate it, and store it all first though.